### ISSUE TITLE
custom UIDs. Permission denied

### ISSUE THREAD

**Issue by okruga** [telepresenceio/telepresence#2524]
**Describe the bug**
Mount /var/run/secrets/kubernetes.io is not created when intercepted deployment configured to use non-default securityContext (custom UID).
traffic-agent init container logs:
```
 kubectl  logs ed-ed-7dd45d7b56-zmc7c -c traffic-agent                                                                                                                                                                                           
2022-04-12 17:47:22.7703 info    Traffic Agent v2.5.5 [pid:1]
2022-04-12 17:47:22.7704 info    {Name:ed-ed Namespace:bb PodIP:10.0.8.33 AgentPort:9900 AppMounts:/tel_app_mounts AppPort:8080 ManagerHost:traffic-manager.ambassador ManagerPort:8081 APIPort:0}
2022-04-12 17:47:22.7709 error   There was a problem with agent mounts: mkdir /tel_app_mounts/var: permission denied
2022-04-12 17:47:22.7817 info    client : Connected to Manager 2.5.5
2022-04-12 17:47:22.7876 debug   Logging at this level "debug"
2022-04-12 17:47:22.7892 debug   client : HandleIntercepts called
```

So, our deployments are using custom UIDs:
```
 securityContext:
     runAsGroup: 65530
     runAsUser: 65530
```
 
 After running `telepresence intercept "deployment with custom UID"` it installs traffic-agent and seems connected:
``` 
intercepted
    Intercept name    : ed-ed-bb
    State             : ACTIVE
    Workload kind     : Deployment
    Destination       : 127.0.0.1:8080
    Volume Mount Point: /var/folders/gk/vlzxhfws35s9r1gzr00z83x1ybfgzp/T/telfs-4038485379
    Intercepting      : all TCP connections
```

But `ls /var/folders/gk/vlzxhfws35s9r1gzr00z83x1ybfgzp/T/telfs-4038485379` have no var folder. In previous versions (I was using 2.5.3) my laptop even got hanged after I tried to ls. But in version 2.5.5 it shows another mount at least, but still no var.

If I remove securityContext specified above from the deployment (just use securityContext: {}) everything works fine.

**To Reproduce**
1. Install telepresence
2. Try to intercept deployment which runs under custom securityContext
3. look at 'kubectl  logs _interceptedpod_ -c traffic-agent  '
4. See error: `There was a problem with agent mounts: mkdir /tel_app_mounts/var: permission denied`

**Expected behavior**
having /var/run/secrets/kubernetes.io/ inside telepresence mount folder.


**Versions (please complete the following information):**
 - Output of `telepresence version`
``` 
telepresence version                                                                                                                                                                                                                                
Client: v2.5.5 (api v3)
Root Daemon: v2.5.5 (api v3)
User Daemon: v2.5.5 (api v3)
```
 - Operating system of workstation running `telepresence` commands
 MacbookPro M1
 - Kubernetes environment and Version [e.g. Minikube, bare metal, Google Kubernetes Engine]
Azure AKS v1.22.2

**VPN-related bugs:**
VPN is not used


**Additional context**
n/a
[telepresence_logs_1204.zip](https://github.com/telepresenceio/telepresence/files/8475828/telepresence_logs_1204.zip)

**Comment by thallgren** [comment]
The logs shows this:
```
2022-04-12 18:43:33.2779 error   There was a problem with agent mounts: mkdir /tel_app_mounts/var: permission denied
```
Not sure why that is. Does this affect just the `var` folder or do you see it for other mounts as well?

**Comment by okruga** [comment]
It seems like in version 2.5.5 it effects `var` folder only, since I can see one other mount.
When I was using version 2.5.3 there were no any mounts available at all.

**Comment by hannaballantine** [comment]
I reproduced the issue using the Quote service through the following steps:
1. Apply the YAML for the quote service and confirm that `telepresence intercept quote --port 8080` creates an intercept. 
Confirm that the logs appear normal and `ls /var/folders/hm/l48yqgms3h58dnr2qrktyznw0000gp/T/telfs-229917217` shows the var folder.
2. Edit the Quote deployment, modifying the securityContext such that it contains a custom UID (I chose runAsGroup: 65534 runAsUser: 65534, I also tested with other UID's and received the same results)
3. Create an intercept for this updated Quote deployment (after leaving the previous intercept).
4. Observe the logs: `kubectl logs interceptedquotesvcpod -c traffic-agent`. 
They show: 
`2022-05-04 18:55:52.7544 info    Traffic Agent v2.5.8 [pid:1]`
`2022-05-04 18:55:52.7545 info    {Name:quote Namespace:default PodIP:10.42.0.16 AgentPort:9900 AppMounts:/tel_app_mounts AppPort:8080 ManagerHost:traffic-manager.ambassador ManagerPort:8081 APIPort:0}`
`2022-05-04 18:55:52.7583 error   There was a problem with agent mounts: mkdir /tel_app_mounts/var: permission denied`

While there appears to be the same error, the difference I observe is that when I intercept with a custom securityContext, the mount does not appear at all (similar to the experience of using 2.5.3), and additionally I receive in my logs following the initial error: `2022-05-04 18:55:52.7584 info    Not starting sftp-server ($APP_MOUNTS is empty or $USER is set)`. Note: I am running on a Mac with an Intel chip and Telepresence v2.5.8.
